src/templates/kit/screens/FundraisingScreen.js

```diff
@@ -0,0 +1,607 @@
+import React, { useCallback, useMemo, useReducer, useState } from 'react'
+import { Decimal } from 'decimal.js'
+import PropTypes from 'prop-types'
+import { Field, GU, Help, Info, Tabs, TextInput, useTheme } from '@aragon/ui'
+import {
+  Header,
+  KnownAppBadge,
+  Navigation,
+  PercentageField,
+  ScreenPropsType,
+} from '..'
+
+const TABS = ['Overview', 'Advanced']
+const INPUT_SMALL = 17 * GU
+const INPUT_MEDIUM = 24 * GU
+
+const DEFAULT_VALUES = {
+  cliffPeriod: 90,
+  expectedGrowth: 200,
+  fundingPeriod: 14,
+  initialPricePerShare: 1.1,
+  initialPricePerShareInput: '1.1',
+  targetGoal: 25000,
+  presalePrice: 1,
+  presalePriceInput: '1',
+  tokensOffered: 90,
+  projectFunding: 10,
+  vestingSchedule: 365,
+  batchLength: 1,
+  slippageDai: 25,
+  slippageAnt: 100,
+  tapRate: 2500,
+  tapFloor: 2500,
+  maximumMonthlyUpdates: 50,
+}
+
+function BN(value) {
+  return new Decimal(value)
+}
+
+function labelDays(value) {
+  return value === 1 ? 'Day' : 'Days'
+}
+
+function labelBatch(value) {
+  return value === 1 ? 'Block' : 'Blocks'
+}
+
+function def(value, defaultValue) {
+  return value === undefined ? defaultValue : value
+}
+
+function intDef(value, defaultValue = 0) {
+  const intValue = parseInt(value, 10)
+  return isNaN(intValue) ? defaultValue : intValue
+}
+
+function floatDef(value, defaultValue = 0) {
+  const floatValue = parseFloat(value)
+  return isNaN(floatValue) ? defaultValue : floatValue
+}
+
+function dataDefaults(data) {
+  return Object.keys(DEFAULT_VALUES).reduce(
+    (_data, fieldName) => ({
+      ..._data,
+      [fieldName]: def(data[fieldName], DEFAULT_VALUES[fieldName]),
+    }),
+    {}
+  )
+}
+
+function useConfigureFields(data) {
+  const [fields, _update] = useReducer(reduceFields, dataDefaults(data))
+
+  // Update a field by using its name and value as parameters
+  const update = useCallback((...params) => _update(params), [])
+
+  // Bind the update function to a callback passing the value, e.g.:
+  //
+  //  <PercentageField
+  //   onChange={bindUpdate('fieldName')}
+  //   value={fieldValue}
+  //  />
+  //
+  const bindUpdate = useCallback(name => value => update(name, value), [update])
+
+  return { fields, update, bindUpdate }
+}
+
+function updateVestingSchedule(fields, value) {
+  const vestingSchedule = Math.max(-1, intDef(value, -1))
```

Comment:
What are the `-1` values used for? I don't think I saw them being handled anywhere.

Reply:
This is how empty values are represented: https://github.com/aragon/aragon/pull/1092/files#diff-35739f07b12eb9af65b4c89f83806e19R374

(This is one of the things that could get improved once we have a better way to handle forms: allowing a field to be empty will be somewhere else.)